kolibri/plugins/user_profile/tasks.py

```diff
@@ -14,17 +18,54 @@
 
 
 class MergeUserValidator(PeerImportSingleSyncJobValidator):
-    local_user_id = HexOnlyUUIDField()
+    local_user_id = serializers.PrimaryKeyRelatedField(
+        queryset=FacilityUser.objects.all()
+    )
+    remote_admin_password = serializers.CharField(default=None, required=False)
+    remote_admin_username = serializers.CharField(default=None, required=False)
 
     def validate(self, data):
+        self.create_remote_user(data)
         job_data = super(MergeUserValidator, self).validate(data)
-        local_user_id = data["local_user_id"]
-        if not FacilityUser.objects.filter(id=local_user_id).exists():
+        job_data["kwargs"]["local_user_id"] = data["local_user_id"].id
+        return job_data
+
+    def create_remote_user(self, data):
+        remote_admin_password = data.get("remote_admin_password")
+        remote_admin_username = data.get("remote_admin_username")
+
+        # create the remote user if remote admin credentials are provided
+        if not remote_admin_password or not remote_admin_username:
+            return
+
+        if "baseurl" not in data and "device_id" not in data:
             raise serializers.ValidationError(
-                "An user with the given id does not exist in this facility"
+                "Either baseurl or device_id must be specified"
             )
-        job_data["kwargs"]["local_user_id"] = local_user_id
-        return job_data
+        if data.get("device_id", None) is not None:
+            if not data["device_id"].base_url:
+                raise serializers.ValidationError("Device has no base url")
+            baseurl = data["device_id"].base_url
+        else:
+            baseurl = data["baseurl"]
+
+        facility = data["facility"]
+        user_data = {
+            "username": data["username"],
+            "password": data["password"],
+            "facility": facility,
+        }
+        create_user_url = urljoin(baseurl, reverse("kolibri:core:facilityuser-list"))
+        auth_data = {
+            "username": remote_admin_username,
+            "password": remote_admin_password,
+        }
+        auth_url = urljoin(baseurl, reverse("kolibri:core:publicsignup-list"))
+        with requests.Session() as req:
+            req.post(auth_url, data=auth_data)
```

Comment:
This is now signing up to the remote facility with the admin username and password - that's not what we want to do here.

We should be POSTing the new user data to the public signup URL, and not using the internal facility user API at all.

Reply:
Ok, I had totally missed the purpose of the new api. 
The change was trivial. I only have doubt if having two different endpoints to create/not create users is better. WIth this implementation it's the same endpoint. If the user does not exist in the remote server it tries to create it using the new api.